app/src/main/java/com/allan/boardbuddies/EditActivity.java

```diff
@@ -21,20 +21,24 @@ public class EditActivity extends AppCompatActivity {
     private String localFilename;
     private String localTitle;
     private String localContent;
+    private int localPosition;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_edit);
         Toolbar toolbar = findViewById(R.id.edit_note_toolbar);
         toolbar.setNavigationOnClickListener(v -> {
+            Intent resultIntent = new Intent();
             if (localFilename == null){ //if there is no local file: saveTextNote()
-                saveTextNote();
+                resultIntent.putExtra("addedFilename", saveTextNote());
             } else if (!localTitle.equals(editTextTitle.getText().toString()) || !localContent.equals(editTextContent.getText().toString())) { //if local file has been changed
-                saveTextNote();
+                resultIntent.putExtra("addedFilename", saveTextNote());
```

Comment:
nit: is it always a newly added file or is something like "modifiedFilename" more accurate?

Reply:
No if the file is modified I'm deleting the old one and making a new one